azext_iot/tests/iothub/devices/test_hub_export_import.py

```diff
@@ -0,0 +1,364 @@
+
+        if(device1["authenticationType"] == DeviceAuthApiType.sas.value):
+            id1 = self.cmd("iot hub device-identity show -l {} -d {}".format(self.connection_string, device1['deviceId'])) \
+                .get_output_in_json()
+            id2 = self.cmd("iot hub device-identity show -l {} -d {}".format(self.dest_hub_cstring, device1['deviceId'])) \
+                .get_output_in_json()
+            assert(id1["authentication"]["symmetricKey"]["primaryKey"] == id2["authentication"]["symmetricKey"]["primaryKey"])
+            assert(id1["authentication"]["symmetricKey"]["secondaryKey"] ==
+                   id2["authentication"]["symmetricKey"]["secondaryKey"])
+
+        if(device1["authenticationType"] == DeviceAuthApiType.selfSigned.value):
+            assert(device1["x509Thumbprint"]["primaryThumbprint"] == device2["x509Thumbprint"]["primaryThumbprint"])
+            assert(device1["x509Thumbprint"]["secondaryThumbprint"] == device2["x509Thumbprint"]["secondaryThumbprint"])
+
+        assert(len(device1["properties"]["desired"]) == len(device2["properties"]["desired"]))
+
+        for prop in device1["properties"]["desired"]:
+            if(prop != "$metadata" and prop != "$version"):
+                assert(prop in device2["properties"]["desired"])
+                assert(device1["properties"]["desired"][prop] == device2["properties"]["desired"][prop])
+
+        # compare modules
+
+        orig_modules = self.cmd(
+            "iot hub module-identity list -d {} -l {}".format(device1['deviceId'], self.connection_string)
+        ).get_output_in_json()
+        dest_modules = self.cmd(
+            "iot hub module-identity list -d {} -l {}".format(device1['deviceId'], self.dest_hub_cstring)
+        ).get_output_in_json()
+
+        if(device1["capabilities"]["iotEdge"]):
+            assert(len(orig_modules) == len(dest_modules) == 3)
+        else:
+            assert(len(orig_modules) == len(dest_modules) == 1)
+
+        for module in orig_modules:
+            target_module = None
+            for mod in dest_modules:
+                if(module["moduleId"] == mod["moduleId"]):
+                    target_module = mod
+                    break
+
+            assert target_module
+
+            assert(module["authentication"] == target_module["authentication"])
+
+            module_twin = self.cmd(
+                f"iot hub module-twin show -m {module['moduleId']} -d {device1['deviceId']} -l {self.connection_string}"
+            ).get_output_in_json()
+            target_module_twin = self.cmd(
+                f"iot hub module-twin show -m {module['moduleId']} -d {device1['deviceId']} -l {self.dest_hub_cstring}"
+            ).get_output_in_json()
+
+            assert(len(module_twin["properties"]["desired"]) == len(target_module_twin["properties"]["desired"]))
+            for prop in module_twin["properties"]["desired"]:
+                if(prop != "$metadata" and prop != "version"):
```

Comment:
prop not in ["$metadata", "version"] would be easier - especially if the service decides to make another default prop